lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -2282,6 +2372,34 @@ static void computeReuseInFunc(func::FuncOp func, VectState *state) {
     computeEnclosingLoopsPerBlock(forOp, state, enclosingLoops);
   }
 
+  WalkResult result = func.walk([&](TransferReadOp op) {
+    if (failed(isUnalignedLoad(op, state))) {
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+
+  if (result.wasInterrupted()) {
+    return failure();
+  }
+
+  return success();
+}
+
+// Compute the reuse interval for all the transfer_read operations. The
+// transfer_read operations capture the vector load. Since AIE only allows for
+// aligned vector loads, we need to compose multiple transfer reads together to
+// form intervals of certain width (128, 256, 512, or 1024), and create an AIE
+// vector from each interval.
+static void computeReuseInFunc(func::FuncOp func, VectState *state) {
+  if (!unalignedLoadsCheck) {
+    for (AffineForOp forOp : func.getOps<AffineForOp>()) {
+      SmallVector<Operation *, 8> enclosingLoops;
+      enclosingLoops.push_back(forOp);
+      computeEnclosingLoopsPerBlock(forOp, state, enclosingLoops);
```

Comment:
Function computeEnclosingLoopsPerBlock() modifies "state".
The original code has only 1 calls site of this function.
The new code has 2 call sites. In case that unalignment-check is enabled computeEnclosingLoopsPerBlock is called "early" during isUnalignedLoad(). Otherwise it is still called in computeReuseInFunc.
This is unfortunate, because it makes the new code harder to follow, and moreover the isUnalignedLoad() function now has the side effect of changing "state".

Reply:
After computeEnclosingLoopsPerBlock(), the only change is state->blockToEnclosingLoops, which was an empty map before computeEnclosingLoopsPerBlock(). I don't think this is a negative side effect. The key thing is we can have an early exit when it hits an unalignment load and we don't need to do any more analysis and transformation. If you think the code between line 2396-2400 is hard to follow, I can just use it once and put it before line 2465.

Reply:
Yes. Moving 2396-2399 to before line 2465 would address my concern.